datafusion/functions/src/math/bounds.rs

```diff
@@ -0,0 +1,137 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use arrow::datatypes::DataType;
+use datafusion_common::ScalarValue;
+use datafusion_expr::interval_arithmetic::Interval;
+
+fn unary_interval_datatype(interval: &Interval) -> DataType {
+    interval.lower().data_type()
+}
+
+pub(super) fn unbounded(_input: &[&Interval]) -> crate::Result<Interval> {
+    // We cannot assume the input datatype is the same of output type.
+    Interval::make_unbounded(&DataType::Null)
+}
```

Comment:
Is it better here returning `not_impl_err`? The current version may cause type mismatch errors and debugging would require more time.

Reply:
I think that would cause an error when trying to use these functions? The update [macro](https://github.com/apache/datafusion/pull/11584/files#diff-b2d8a351fea4f0f656b5535ab00541f6e413eba67d8a78f798f80cfb440bbb32) is going to call whatever is passed as `$EVALUATE_BOUNDS`. Right now this implementation matches the default of the `ScalarUDFImpl` trait.

I think an alternative is to make something like `Interval::make_symmetric_inf`. I'm using unbounded here, but probably in most of these cases it should be (-inf, inf) which could be given a type.

Reply:
Went ahead and did that here: [cf2b023](https://github.com/apache/datafusion/pull/11584/commits/cf2b023ac61196f878e0078ee54250f2a586aaad)